presto-main/src/main/java/com/facebook/presto/server/PrestoServer.java

```diff
@@ -138,6 +141,10 @@ public void run()
                     injector.getInstance(ServerConfig.class),
                     injector.getInstance(NodeSchedulerConfig.class));
 
+            updateThriftServerPort(
```

Comment:
Add a TODO that thrift server port should be announced by discovery server (similar to http/https port, we already forked airlift:discovery so it's do-able 😃 ).  Maybe @jessesleeping  can work on this next half?